llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -1378,6 +1380,27 @@ class CastInst : public Instruction {
 #endif
 };
 
+class SIToFPInst final : public CastInst {
```

Comment:
I wonder if it's possible to do some CRTP magic to stamp out all of these instantiations much more succinctly. Let me mess around locally.

Reply:
Yeah that works. Do you want to create a PR for this?

Reply:
I could also try doing something similar for the tests, since there is a lot of repetition there too. The only downside is that if there is a failure the error reported won't be as easy to follow, but I doubt these tests will fail.

If we go this way, then the rest of the CastInst sub-classes can be a single PR which would be really nice.

Wdyt?

Reply:
https://github.com/llvm/llvm-project/pull/101427